index.js

```diff
@@ -289,7 +303,17 @@ const parseArgs = (config = kEmptyObject) => {
       validateObject(optionConfig, `options.${longOption}`);
 
       // type is required
-      validateUnion(objectGetOwn(optionConfig, 'type'), `options.${longOption}.type`, ['string', 'boolean']);
+      const optionType = objectGetOwn(optionConfig, 'type');
+      validateUnion(optionType, `options.${longOption}.type`, ['string', 'boolean']);
+
+      if (ObjectHasOwn(optionConfig, 'defaultValue')) {
+        const defaultValue = objectGetOwn(optionConfig, 'defaultValue');
+        if (optionType === 'string') {
+          validateString(defaultValue, `options.${longOption}.defaultValue`);
+        } else if (optionType === 'boolean') {
+          validateBoolean(defaultValue, `options.${longOption}.defaultValue`);
+        }
+      }
```

Comment:
Thoughts on additionally factoring in `optionConfig.multiple`? For example if `multiple` is `true`, validate the default value is an array and that each element aligns with the `optionType`.

Reply:
Currently as a side-affect of the storage, a default value of `str` will get stored as a value of `['str`]` if the value is "multiple". This makes the default value look like what the user would enter on the command-line (for a string), but the default value does not look like the resulting value and does not allow an empty array or an array with more than one element.

I am not sure there is a right answer. I somewhat prefer the idea the default value matches the final format. But this is not consistent with the idea that the default value would go through (theoretical) validation or type coercion.

For interest, Commander's behaviour from early days is that default values are directly stored as specified and do not go through type coercion.

Reply:
I think the behaviour @aaronccasanova suggested is pretty reasonable: https://github.com/pkgjs/parseargs/pull/142#discussion_r945328537

Reply:
> I somewhat prefer the idea the default value matches the final format.

This is my preference as well. IMO this is more intuitive for authors/readers and should make for a very simple update to the typedefs.

Here's a trivial example of what I would expect (which is not possible in its current state):
```js
const cli = parseArgs({ options: { verbose: { type: 'boolean', default: [true, true] } } })
console.log(cli.values.verbose.length)

// node foo.js -vvv
// 3

// node foo.js
// 2
```

Note: With this update I would also expect an empty array to fail validation as I don't believe that scenario would ever occur in the `parseArgs` results.

Reply:
> Note: With this update I would also expect an empty array to fail validation as I don't believe that scenario would ever occur in the parseArgs results.

Interesting. Two cases that a user might try to use are default empty array for a multiple, and default `false` for a type-boolean option. Neither are possible results of argument parsing, but are both giving the option a suitable "defined" value rather than it being undefined. I was expecting both would be allowed.

Reply:
Oop! I missed a critical option config in my above example (`multiple: true`). Sorry if that caused any confusion e.g.

```js
const cli = parseArgs({
  options: { verbose: { type: 'boolean', multiple: true, default: [true, true] } }
})
```

> Two cases that a user might try to use are default empty array for a multiple, and default `false` for a type-boolean option. Neither are possible results of argument parsing...


While those two cases are consistent with our [typedefs](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/279d05c4d223fad9455ceed99bd136c974cb4c17/types/node/util.d.ts#L1199-L1202) (e.g. allowing and empty array when `multiple: true` and `false` when `type: 'boolean'` is set), the mismatch in behavior from the `parseArgs` results could be confusing. This opinion is very very loosely held. The main addition I think would be valuable is validating each default value aligns to the typedefs of the parse results.

Reply:
If we're going to do this at all, I agree with @shadowspawn that supporting `false` for boolean options and empty arrays for `multiple` options is good. I'm not really worried about potential for confusion since the user explicitly asked for it..

Reply:
Another example might be, putting a non-array as the default for a multiple type, or putting a non-boolean as the default for a boolean type.

Reply:
I have added this feature.

Now:

- when type `boolean` and `multiple: false` = only `boolean` accepted
- when type `boolean` and `multiple: true` = only array of `boolean` accepted (empty array is fine)
- when type `string` and `multiple: false` = only `string` accepted
- when type `string` and `multiple: true` = only array of `string` accepted (empty array is fine)

Reply:
The described behavior looks great! However, there may be an issue with the logic: https://github.com/pkgjs/parseargs/pull/142#discussion_r947169597

Reply:
I'm happy with the updated validation behavior, thanks @Eomm!

Going to resolve this conversion now, but feel free (anyone) to reopen or start a new discussion regarding the changes.